### PR TITLE
[stable/redis-ha] Add a dedicated port in haproxy for redis slaves

### PR DESCRIPTION
**Pull request by amalsom10**
* [stable/redis-ha] Add a dedicated port in haproxy for redis slaves

Signed-off-by: Amal Soman <amalsoman10@gmail.com>

#### What this PR does / why we need it:
  - While running this is production got a new requirement of Read-only endpoint point to Redis slave from dev side. So update the hap config to expose a new port 6380 to point to slaves.
  - The service of hap will be same but added one more port to expose slave.
  - Similar to the read-only endpoint of Elasticcache in AWS.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #
  - The requirement of the read-only port in hap for Redis-slaves

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by Raab70** [comment]
Fixes #16824

File: stable/redis-ha/values.yaml [hunk 1]
```diff
@@ -51,6 +51,11 @@ haproxy:
   init:
     resources: {}
 
+ # Enable if you want a dedicated port in haproxy for redis-slaves
+hapreadport:
```
Comment (ssalaues) [inline stable/redis-ha/values.yaml]: I don't think this necessarily needs it's own section and makes more sense if it falls under the `haproxy` yaml block. Something like:
```
haproxy:
  enabled: false
  readOnlyPort:
    enabled: true
    port: 6380
```

Comment (amalsom10) [reply stable/redis-ha/values.yaml]: sure, will do the changes and commit again. Thanks

Comment (amalsom10) [reply stable/redis-ha/values.yaml]: @ssalaues updated the template with required changes

File: stable/redis-ha/values.yaml [hunk 2]
```diff
@@ -26,6 +26,10 @@ serviceAccount:
 ## ref: https://cbonte.github.io/haproxy-dconv/1.9/intro.html
 haproxy:
   enabled: false
+  # Enable if you want a dedicated port in haproxy for redis-slaves
+  readOnlyPort:
```
Comment (DandyDeveloper) [inline stable/redis-ha/values.yaml]: I think renaming this to `slaveEndpoint` or `readOnly` makes more sense. As it's likely we'd want to add stuff in future for read only scenarios.

**Review by DandyDeveloper — CHANGES_REQUESTED** [review]
Looks good and I can see the inherent benefit. 

Just one minor change, similar to Salim but I want to allow the ability to extend on this in future.

File: stable/redis-ha/values.yaml (hunk 2 above)
Comment (amalsom10) [reply stable/redis-ha/values.yaml]: @DandyDeveloper @ssalaues should I rename the same to `readOnly`

**Review by amalsom10 — COMMENTED** [review]
@DandyDeveloper great thought. Changed the value to `readOnly`

File: stable/redis-ha/values.yaml (hunk 2 above)
Comment (amalsom10) [reply stable/redis-ha/values.yaml]: great thought. Changed the value to readOnly

**Pull request merged**